core/types/transaction.go

```diff
@@ -205,24 +199,24 @@ func UnmarshalTransactionFromBinary(data []byte) (Transaction, error) {
 	default:
 		if data[0] >= 0x80 {
 			// Tx is type legacy which is RLP encoded
-			return DecodeTransaction(data)
+			s := rlp.NewStream(bytes.NewReader(data), uint64(len(data)))
+			return DecodeRLPTransaction(s)
 		}
 		return nil, ErrTxTypeNotSupported
 	}
 }
 
-func UnmarshalWrappedTransactionFromBinary(data []byte) (Transaction, error) {
-	if len(data) <= 1 {
-		return nil, fmt.Errorf("short input: %v", len(data))
-	}
-	if data[0] != BlobTxType {
-		return UnmarshalTransactionFromBinary(data)
-	}
-	t := &BlobTxWrapper{}
-	if err := DecodeSSZ(data[1:], t); err != nil {
-		return nil, err
+func unmarshalWrappedTransactionFromBinary(data []byte) (Transaction, error) {
+	switch data[0] {
```

Comment:
since you removed the length check, you should document the precondition here that data should be at least 1 byte.

Reply:
Also: My personal preference is to avoid switches for checking a single value as I think it reduces readability, but maybe thiis is idiomatic in Erigon?